packages/vscode/src/plugins/nextjs/utils.ts

```diff
@@ -0,0 +1,24 @@
+import path from 'path'
+import { SourceFile } from 'ts-morph'
+import { NextFunctionName, NextFunctions, NextFunctionType } from './constants'
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function isJS(sourceFile: SourceFile) {
+  const filePath = sourceFile.getFilePath()
+  const extension = path.extname(filePath)
+  return extension.includes('js')
+}
+
+export function getUsedNextFunctions(sourceFile: SourceFile): NextFunctionType {
```

Comment:
Can you please add a comment about what this function does? 
All cases in which objects are undefined are not handled at all. Is this expected?

Reply:
Reads a source file and finds which special nextJS functions are being used. i.e `getServerSideProps` | `getStaticProps`